Library/Homebrew/download_strategy.rb

```diff
@@ -832,7 +832,12 @@ def commit_outdated?(commit)
     else
       return true unless commit
       return true unless @last_commit.start_with?(commit)
-      multiple_short_commits_exist?(commit)
+      if multiple_short_commits_exist?(commit)
+        true
+      else
+        version.update_commit(commit)
+        false
+      end
```

Comment:
This tries to shrink the commit for the version if it's possible. When we pass the commit to this method, we can fetch a full-length comment and overwrite short one with full-length for the formula. If there is no multiple short commits, there is no need to keep it full-length and we update it with initial one.

`commit_outdated?` is called here: https://github.com/Homebrew/brew/blob/master/Library/Homebrew/formula.rb#L522